lib/souyuz/runner.rb

```diff
@@ -1,23 +1,90 @@
 require 'fastlane'
+require 'provision_util'
 
 module Souyuz
   class Runner
     def run
       config = Souyuz.config
 
-      build_app
-
       if Souyuz.project.ios? or Souyuz.project.osx?
+
+        configuration = "'#{config[:build_configuration]}|#{config[:build_platform]}'"
+
+        # Set provisioning profile
+        if config[:provision_profile_uuid]
+          provision_profile_default_path = "#{Dir.home}/Library/MobileDevice/Provisioning\ Profiles/".freeze
+          provision_profile_path = File.join provision_profile_default_path, "#{config[:provision_profile_uuid]}.mobileprovision"
+          cert = ProvisionUtil::get_cert_from_provision(provision_profile_path)
+          subject = Hash[cert.subject.to_a.map {|arr|  [arr[0], arr[1]]}]
+          # config[:provision_profile_uuid] -> CodesignProvision
+          name = subject['CN'] # CodesignKey
+
+          file = File.new(Souyuz.config[:project_path])
+          doc = Nokogiri::XML(file.read)
+          file.close
+
+          [
+            ['CodesignProvision', config[:provision_profile_uuid]],
+            ['CodesignKey', name]
+          ].each do |dict|
+            doc.search('PropertyGroup').each do |group|
+              next unless !group['Condition'].nil? && group['Condition'].include?(configuration)
+              property = group.search(dict[0])
+              if property.size > 0
+                property[0].content = dict[1]
+              end
+            end
+          end
+
+
+          xml = doc.to_xml
+          UI.command_output(xml) if $verbose
+          File.write(Souyuz.config[:project_path], xml)
+        end
+        
+        # Add "ArchiveOnBuild"
+        if config[:archive_app] == true
+          file = File.new(Souyuz.config[:project_path])
+          doc = Nokogiri::XML(file.read)
+          file.close
+
+          doc.search('PropertyGroup').each do |group|
+            next unless !group['Condition'].nil? && group['Condition'].include?(configuration)
+            archiveNode = Nokogiri::XML::Node.new "ArchiveOnBuild", doc
+            archiveNode.content = config[:archive_app]
+            group.add_child(archiveNode)
+          end
+
+
+          xml = doc.to_xml
+          UI.command_output(xml) if $verbose
+          File.write(Souyuz.config[:project_path], xml)
```

Comment:
I think it should be noted for the user, that using this function will change the project file. That could be an issue for some CI setups. So I think its best to make the user aware of that.